pkg/detectors/postgres/postgres.go

```diff
@@ -0,0 +1,170 @@
+    }
+    defer db.Close()
+
+    ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+    defer cancel()
+
+
+    // Try to establish a connection
+    err = db.PingContext(ctx)
+    if err != nil {
+        return err
+    }
+
+    // If we reach here, the credentials are valid
+    return nil
+}
+
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	// Check for inline connection strings
+	uriMatches := uriPat.FindAllString(dataStr, -1)
+	for _, uri := range uriMatches {
+		pgURL, err := url.Parse(uri)
+		if err != nil {
+			continue
+		}
+
+		// PostgreSQL URLs might not always have the userinfo (username:password) part
+		if pgURL.User != nil {
+			username := pgURL.User.Username()
+			password, _ := pgURL.User.Password()
+			hostname := pgURL.Hostname()
+			s1 := detectors.Result{
+				DetectorType: detectorspb.DetectorType_Postgres,
+				Raw:          []byte(strings.Join([]string{hostname, username, password}, "\t")),
+			}
+            if verify {
+                verificationErr:= verifyPostgres(pgURL)
+                s1.Verified = verificationErr == nil
+            }
+			results = append(results, s1)
+		}
+	}
+
+	// Check for separate components
+    usernameRegexState := common.UsernameRegexCheck("")
+	usernameMatches := usernameRegexState.Matches(data)
+
+	passwordRegexState := common.PasswordRegexCheck("") // No explicit character exclusions by Snowflake for passwords
+	passwordMatches := passwordRegexState.Matches(data)
+	hostnameMatches := hostnamePat.FindAllStringSubmatch(dataStr, -1)
+
+    // Combine the separate components into potential credentials
+    for _, username := range usernameMatches {
+        if len(username) < 2 {
+            continue
+        }
+        for _, hostname := range hostnameMatches {
+            if len(hostname) < 2 {
+                continue
+            }
+            result := false
+            s1 := detectors.Result{
+                DetectorType: detectorspb.DetectorType_Postgres,
+            }
+            for _, password := range passwordMatches {
+                if len(password) < 2 {
+                    continue
+                }
+                
+                // Since we're combining these, we should probably also ensure that the total length does not exceed the 255 character limit for hostnames
+                combinedLength := len(username) + len(password) + len(hostname[1])
+                if combinedLength > 255 {
+                    continue // Skip if the combined length is too long
+                }
+                s1.Raw = []byte(strings.Join([]string{hostname[1], username, password}, "\t"))
+                result = true
+                postgresURL := url.URL{
+                    Scheme: "postgresql",
+                    User:   url.UserPassword(username, password),
+                    Host:   fmt.Sprintf("%s:%s", hostname[1], "5432"),
```

Comment:
In the verify function there is handling for the default port if nothing else was specified, but it looks like this code always assumes a default port of "5432"